tokens/cnft-vault/app/withdrawWithLookup.ts

```diff
@@ -0,0 +1,169 @@
+  const asset2 = await getAsset(assetId2);
+
+  const proof1 = await getAssetProof(assetId1);
+  const proofPathAsAccounts1 = mapProof(proof1);
+  const proof2 = await getAssetProof(assetId2);
+  const proofPathAsAccounts2 = mapProof(proof2);
+
+  const ixData1 = getInstructionData(asset1, proof1);
+  const ixData2 = getInstructionData(asset2, proof2);
+
+  const remainingAccounts: AccountMeta[] = [...proofPathAsAccounts1, ...proofPathAsAccounts2];
+
+  const ix = await program.methods.withdrawTwoCnfts(...ixData1, ...ixData2)
+    .accounts({
+      leafOwner: vaultPDA,
+      merkleTree1: tree1,
+      newLeafOwner1: receiver1,
+      treeAuthority1: treeAuthority1,
+      merkleTree2: tree2,
+      newLeafOwner2: receiver2,
+      treeAuthority2: treeAuthority2,
+      bubblegumProgram: BUBBLEGUM_PROGRAM_ID,
+      compressionProgram: SPL_ACCOUNT_COMPRESSION_PROGRAM_ID,
+      logWrapper: SPL_NOOP_PROGRAM_ID,
+      systemProgram: anchor.web3.SystemProgram.programId
+    })
+    .remainingAccounts(remainingAccounts)
+    .instruction();
+
+
+  await extendLookupTable(lookupTable, proofPathAsAccounts1.map(acc => acc.pubkey));
+  await extendLookupTable(lookupTable, proofPathAsAccounts2.map(acc => acc.pubkey));
+
+  const lookupTableAccount = await connection
+    .getAddressLookupTable(lookupTable)
+    .then((res) => res.value);
+
+  if (!lookupTableAccount) {
+    console.log("could not fetch ATL!");
+    return;
+  }
+
+  await new Promise(_ => setTimeout(_, 30000));
+
+  const messageV0 = new TransactionMessage({
+    payerKey: keypair.publicKey,
+    recentBlockhash: (await connection.getLatestBlockhash()).blockhash,
+    instructions: [ix],
+  }).compileToV0Message([lookupTableAccount]);
+
+  const transactionV0 = new VersionedTransaction(messageV0);
+  transactionV0.sign([keypair]);
+
+  const txid = await connection.sendTransaction(transactionV0);
+  console.log(txid);
+};
+
+function getInstructionData(asset: any, proof: any):
+  [number[], number[], number[], anchor.BN, number, number] {
+  const root = decode(proof.root);
+  const dataHash = decode(asset.compression.data_hash);
+  const creatorHash = decode(asset.compression.creator_hash);
+  const nonce = new anchor.BN(asset.compression.leaf_id);
+  const index = asset.compression.leaf_id;
+  const proofLength = proof.proof.length;
+  return [root, dataHash, creatorHash, nonce, index, proofLength];
+}
+
+main();
+
+async function extendLookupTable(lookupTableAddress: PublicKey, proofHashes: PublicKey[]) {
+
+  const extendInstruction = AddressLookupTableProgram.extendLookupTable({
+    payer: keypair.publicKey,
+    authority: keypair.publicKey,
+    lookupTable: lookupTableAddress,
+    addresses: [
+      ...proofHashes
+    ],
+  });
+
+  const tx = new Transaction();
+  tx.add(extendInstruction);
+
+  const sx = await sendAndConfirmTransaction(connection, tx, [keypair], { commitment: "finalized" });
+  console.log(sx);
+  console.log("ALT extended!");

```

Comment:
Do atls need to be finalized?

Reply:
no, but it's a timing thing. if I wait for finalized that (most likely)  the next read will have up-to-date information.